### ISSUE TITLE
Add type signatures to index overview

### ISSUE THREAD

**Issue by ghc-mirror** [haskell/haddock#261]
**Original reporter**: _dluposchainsky@<hidden>_

I've recently spent a lot of time in the GHC source and was clicking through names in the index just to find out the types for certain identifiers. Paired with the large amount of empty space the index table has, I thought it would be a good idea to fill that empty space with type signatures. The index would then have three columns: name, defining module, type. (For types/newtypes the actual definition could be filled in instead.) What do you think?

**Comment by expipiplus1** [comment]
This would be very useful sometimes!

**Comment by Fuuzetsu** [comment]
Yep. But at least we have quick jump these days that at least eases the pain.
